src/openms/source/ANALYSIS/MAPMATCHING/TransformationModel.cpp

```diff
@@ -60,5 +63,295 @@ namespace OpenMS
   {
     params.clear();
   }
+  
+  void TransformationModel::weightData(TransformationModel::DataPoints& data, const Param& params)
+  {
+    // weight x values 
+    std::vector<std::string> valid_weights;
+    valid_weights = getValidXWeights();
+    if (params.exists("x_weight") && checkValidWeight(params.getValue("x_weight"), valid_weights) && !data.empty())
+    {
+      std::string x_weight = params.getValue("x_weight");
```

Comment:
I think you could move this line out of the if block. 
This would also make the if clause simpler.
e.g.:
const std::string x_weight = params.exists("x_weight") ? params.getValue("x_weight") : "";
if (!x_weight.empty() && checkValidWeight(x_weight, valid_weights) && !data.empty())